code-challenges.test.js

```diff
@@ -10,43 +10,169 @@
+  return arr.filter((str) => str.toLowerCase().includes(ltr));
+};
+
+//test passed
+
+// ===================================== CHALLENGE #3 =================================================
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("findFullHouse", () => {
+  it("a function that takes in an array of 5 numbers and determines whether or not the array is a 'full house'. A full house is exactly one pair and one three of a kind.", () => {
+    const hand1 = [5, 5, 5, 3, 3];
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4];
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4];
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7];
+    // Expected output: true
+    const hand5 = [1, 2, 7, 2, 7];
+    // Expected output: false
+    const hand6 = [4, 5, 6, 4, 5];
+    // Expected output: false
+    const hand7 = [2, 7, 7, 7, 7];
+    // Expected output: false
+    const hand8 = [8, 8, 8, 8, 8];
+    // Expected output: false
+    const hand9 = [2, 8, 8, 8, 2];
+    // Expected output: true
+    expect(findFullHouse(hand1)).toEqual(true);
+    expect(findFullHouse(hand2)).toEqual(false);
+    expect(findFullHouse(hand3)).toEqual(false);
+    expect(findFullHouse(hand4)).toEqual(true);
+    expect(findFullHouse(hand5)).toEqual(false);
+    expect(findFullHouse(hand6)).toEqual(false);
+    expect(findFullHouse(hand7)).toEqual(false);
+    expect(findFullHouse(hand8)).toEqual(false);
+    expect(findFullHouse(hand9)).toEqual(true);
+  });
+});
+
+//ReferenceError: findFullHouse is not defined
 
 // b) Create the function that makes the test pass.
+
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+// our function will receive an array of numbers
+// we can utilize new Set object since we are only looking for a Set with two values to start with. Everything else (over and below size 2) will not be considered since full house contains only two different numbers.
+// once we pass the Set test, we only need to check for cases where we have one number 4 times. We can test our Set values and filter our array
+//If either comes back more than 3 times (return array.length), the condition will return "false", everything else will return "true"
+
+const findFullHouse = (cards) => {
+  let cardsFound = new Set(cards);
+  const [card1, card2] = cardsFound;
+  if (cardsFound.size != 2) return false;
+  return cards.filter((e) => e === card1).length > 3 ||
+    cards.filter((e) => e === card2).length > 3
+    ? false
+    : true;
+};
```

Comment:
Nice work!  This is a pretty unique approach, but you did a great job documenting your process on this.  👏